### PR TITLE
Upgrade plugin dependencies

### PR DESCRIPTION
**Pull request by chelseakomlo**
Upgraded dependencies necessary for future `go-plugin` work. Also adds a necessary local dev dependency for generating protobuf files.

### CODE REVIEW

File: vendor/vendor.json [hunk 1]
```diff
@@ -2,366 +2,2207 @@
...
-		{"path":"google.golang.org/grpc/status","checksumSHA1":"DIv9qbApAoh2cF2G3Br24lVPqUI=","revision":"0c41876308d45bc82e587965971e28be659a1aca","revisionTime":"2017-07-21T17:58:12Z"},
-		{"path":"google.golang.org/grpc/tap","checksumSHA1":"aixGx/Kd0cj9ZlZHacpHe3XgMQ4=","revision":"0c41876308d45bc82e587965971e28be659a1aca","revisionTime":"2017-07-21T17:58:12Z"},
-		{"path":"google.golang.org/grpc/transport","checksumSHA1":"oFGr0JoquaPGVnV86fVL8MVTc3A=","revision":"0c41876308d45bc82e587965971e28be659a1aca","revisionTime":"2017-07-21T17:58:12Z"},
-		{"path":"gopkg.in/fsnotify.v1","checksumSHA1":"eIhF+hmL/XZhzTiAwhLD0M65vlY=","revision":"629574ca2a5df945712d3079857300b5e4da0236","revisionTime":"2016-10-11T02:33:12Z"},
-		{"path":"gopkg.in/inf.v0","checksumSHA1":"6f8MEU31llHM1sLM/GGH4/Qxu0A=","revision":"3887ee99ecf07df5b447e9b00d9c0b2adaa9f3e4","revisionTime":"2015-09-11T12:57:57Z"},
-		{"path":"gopkg.in/lxc/go-lxc.v2","checksumSHA1":"i97goLq3AIfUNB8l1hxGGMSW0+s=","revision":"f8a6938e600c634232eeef79dc04a1226f73a88b","revisionTime":"2016-08-03T16:52:18Z"},
-		{"path":"gopkg.in/to
```
Comment (dadgar) [inline vendor/vendor.json]: make vendorfmt

File: GNUmakefile [hunk 2]
```diff
@@ -152,6 +152,7 @@ deps:  ## Install build and development dependencies
 	go get -u github.com/elazarl/go-bindata-assetfs/...
 	go get -u github.com/a8m/tree/cmd/tree
 	go get -u github.com/magiconair/vendorfmt/cmd/vendorfmt
+	go get -u github.com/golang/protobuf/protoc-gen-go
```
Comment (dadgar) [inline GNUmakefile]: Can you add a link to the README for development that you need to install protoc: https://github.com/google/protobuf

Comment (dadgar) [reply GNUmakefile]: This would be under developing locally

File: vendor/vendor.json [hunk 3]
```diff
@@ -313,7 +316,7 @@
 		{"path":"golang.org/x/net/lex/httplex","checksumSHA1":"3xyuaSNmClqG4YWC7g0isQIbUTc=","revision":"ab5485076ff3407ad2d02db054635913f017b0ed","revisionTime":"2017-07-19T21:11:51Z"},
 		{"path":"golang.org/x/net/trace","checksumSHA1":"u/r66lwYfgg682u5hZG7/E7+VCY=","revision":"ab5485076ff3407ad2d02db054635913f017b0ed","revisionTime":"2017-07-19T21:11:51Z"},
 		{"path":"golang.org/x/sync/errgroup","checksumSHA1":"S0DP7Pn7sZUmXc55IzZnNvERu6s=","revision":"316e794f7b5e3df4e95175a45a5fb8b12f85cb4f","revisionTime":"2016-07-15T18:54:39Z"},
-		{"path":"golang.org/x/sys/unix","checksumSHA1":"SIhhNXud9FKua0s3WlGPRRkDKRg=","revision":"8b4580aae2a0dd0c231a45d3ccb8434ff533b840","revisionTime":"2017-11-30T16:26:51Z"},
+		{"path":"golang.org/x/sys/unix","checksumSHA1":"su2QDjUzrUO0JnOH9m0cNg0QqsM=","revision":"ac767d655b305d4e9612f5f6e33120b9176c4ad4","revisionTime":"2018-07-08T03:57:06Z"},
```
Comment (schmichael) [inline vendor/vendor.json]: I updated this dep in #4553 as well, but I made sure to vendor the same revision as you so it shouldn't matter who merges first.

Comment (chelseakomlo) [reply vendor/vendor.json]: Thank you! That is helpful.

**Pull request merged**